backend/models/sequence.js

```diff
@@ -20,101 +20,173 @@ const db = require("../handler/db");
-		const activity = await dbCol.findOne({"_id": utils.stringToUUID(activityId)});
+	return {legend: {}};
+};
 
-		if (!activity) {
-			return Promise.reject(responseCodes.ACTIVITY_NOT_FOUND);
-		}
+const Sequence = {};
 
-		this.cleanActivityDetail(activity);
+Sequence.getSequenceActivityDetail = async (account, model, activityId) => {
+	const activity = await db.findOne(account, activityCol(model), {"_id": utils.stringToUUID(activityId)});
 
-		return activity;
+	if (!activity) {
+		throw responseCodes.ACTIVITY_NOT_FOUND;
 	}
 
-	async getSequenceActivities(account, model, sequenceId) {
-		return FileRef.getSequenceActivitiesFile(account, model, utils.uuidToString(sequenceId));
-	}
+	return cleanActivityDetail(activity);
+};
 
-	getSequenceState(account, model, stateId) {
-		return FileRef.getSequenceStateFile(account, model, stateId);
+Sequence.getSequenceActivities = async (account, model, sequenceId) => {
+	return FileRef.getSequenceActivitiesFile(account, model, utils.uuidToString(sequenceId));
+};
+
+Sequence.getSequenceState = async (account, model, stateId) => {
+	return FileRef.getSequenceStateFile(account, model, stateId);
+};
+
+Sequence.getList = async (account, model, branch, revision, cleanResponse = false) => {
+	const history = await History.getHistory(account, model, branch, revision);
+
+	if(!history) {
+		throw responseCodes.INVALID_TAG_NAME;
 	}
 
-	async getList(account, model, branch, revision, cleanResponse = false) {
-		const history = await History.getHistory(account, model, branch, revision);
-		let submodels = [];
+	const refNodes = await getRefNodes(account, model, branch, revision, {project:1});
+	const submodels = refNodes.map(r => r.project);
 
-		if (history.current) {
-			submodels = await getRefNodes(account, model, branch, revision, {project:1});
-			submodels = submodels.map(r => r.project);
+	const submodelSequencesPromises = Promise.all(submodels.map((submodel) => Sequence.getList(account, submodel, "master", null, cleanResponse)));
+
+	const sequences = await db.find(account, sequenceCol(model), {"rev_id": history._id});
+	sequences.forEach((sequence) => {
+		sequence.teamspace = account;
+		sequence.model = model;
+
+		if (cleanResponse) {
+			cleanSequenceList(sequence);
 		}
+	});
 
-		const submodelSequencesPromises = Promise.all(submodels.map((submodel) => this.getList(account, submodel, "master", null, cleanResponse)));
+	const submodelSequences = await submodelSequencesPromises;
+	submodelSequences.forEach((s) => sequences.push(...s));
 
-		const dbCol = await db.getCollection(account, model + ".sequences");
+	return sequences;
+};
 
-		const sequences = await (dbCol.find({"rev_id": history._id}).toArray());
-		sequences.forEach((sequence) => {
-			sequence.teamspace = account;
-			sequence.model = model;
+Sequence.updateSequence = async (account, model, sequenceId, data) => {
+	if (data && data.name && utils.isString(data.name) && data.name !== "" && data.name.length < 30) {
```

Comment:
I think this is a little clearer
```suggestion
	if (!data || !data.name || !utils.isString(data.name) || data.name == ""  || data.name.length >= 30) {
		throw responseCodes.INVALID_ARGUMENTS;
	}
```

and get rid of the else part